CHANGELOG.md

```diff
@@ -19,6 +19,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Moved and renamed target transform utility functions
```

Comment:
maaan @AdrianSosic how can we reduce your **insane** appetite for refactorings? You know, these 35+ files-changed PRs are not only draining for the coder but perhaps even more for the reviewer...

dont get me wrong, nothing has been made worse in this PR, but I felt like a good chunk of changes in here were just alternatives of preference for something instead of actual improvement (i know youll disagree with it)

but somehow this PR was only made for having a basis for the simulation user guide, just renaming the names of the three main sim functions to make them consistent as discussed would have sufficed imo...

Reply:
I agree with Martin here. Having these PRs with massive changes is really challenging. I do see the point that after a refactoring, it is somehow necessary to change stuff everywhere, but we might be able to make this on a more atomic level (Maybe as follows: Have one PR just initializing the new package without changing the dependencies; keep this open, start another PR on top of this which then does the refactoring of imports; merge these two together somehow - don't know if this works however).

Reply:
Point taken and accepted 😄 I also have to apologize, this went a bit out of hand. But you know, especially signature changes etc. often come with many file changes =/ ... So next time, I will absolutely present that stuff to you in way smaller chunks. But one thing I cannot apologize for is the refactoring itself – you know, the simulation module has really been a pain ever since and I've already done so many refactorings on it without which advanced stuff like TL would have become an absolute nightmare to implement. So 1) yes, smaller chunks absolutely but 2) refactoring here is not yet over 😄 though next time I'll ask for approval before I start working on it 👍🏼